spring-integration-file/src/main/java/org/springframework/integration/file/dsl/FileInboundChannelAdapterSpec.java

```diff
@@ -75,6 +76,22 @@ FileInboundChannelAdapterSpec directory(File directory) {
 		return _this();
 	}
 
+	/**
+	 * A convenient flag to determine if target message source should use a
+	 * {@link RecursiveDirectoryScanner} or stay with a default one.
+	 * @param recursive to set or not a {@link RecursiveDirectoryScanner}.
+	 * @return the spec.
+	 * @see org.springframework.integration.file.RecursiveDirectoryScanner
+	 * @since 5.5
+	 */
+	public FileInboundChannelAdapterSpec recursive(boolean recursive) {
+		if (recursive) {
+			new DirectFieldAccessor(this.target).setPropertyValue("scanner", new RecursiveDirectoryScanner());
```

Comment:
Why use recursion when target has `setScanner`?

Reply:
I believe you mean "reflection".
It is because the `setFilter` and `setLocker` check for the external scanner. See `FileReadingMessageSource.scannerExplicitlySet` property and its usage.
According end-user experience it is a bit cumbersome to move all the fluent filter configuration to the external scanner, when they just want to change the default scanner to recursive one.

I think we can find some other way to do that with some new setter, but I don't think it hurts at the moment to do this reflection once on configuration phase. If you insist, I'll revise it anyway.

Reply:
No that's fine; yes, I meant reflection 😄

Reply:
OK. Let me push some fix and we will continue discussion if that!